apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/ConsumerFactory.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright 2016 Google Inc. All Rights Reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Api.Gax;
+
+namespace Google.Cloud.Diagnostics.Common
+{
+    internal static class ConsumerFactory<T>
+    {
+        /// <summary>
+        /// Gets a <see cref="IConsumer{T}"/> that wraps the passed in consumer based on the buffer options.
+        /// </summary>
+        /// <param name="consumer">The consumer to buffer <see cref="T"/> into.</param>
+        /// <param name="sizer">A sizer for the type</param>
+        /// <param name="options">Buffer options for the buffered consumer</param>
+        internal static IConsumer<T> GetConsumer(IConsumer<T> consumer, ISizer<T> sizer, BufferOptions options)
```

Comment:
Since this is internal, it might be simpler to just be an instance method on BufferOptions itself, but perhaps your intention is just to make BufferOptions and simple class.